When calico-felix dies with an error, it should not return success

calico-felix is not doing the right thing:

```
# calico-felix ; echo $?
2014-12-19 17:49:14,069 [ERROR] calico.felix.felix 728: Felix exiting after uncaught exception
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/calico/felix/felix.py", line 724, in main
    agent = FelixAgent(config_path)
  File "/usr/lib64/python2.7/site-packages/calico/felix/felix.py", line 59, in __init__
    self.config = Config(config_path)
  File "/usr/lib64/python2.7/site-packages/calico/felix/config.py", line 63, in __init__
    500))
  File "/usr/lib64/python2.7/site-packages/calico/felix/config.py", line 133, in get_cfg_entry
    section, self._config_path)
ConfigException: Section global missing from config file (file path : felix.cfg)
0
```

The acl manager and agent are exiting correctly:

```
# calico-acl-manager ; echo $?
Traceback (most recent call last):
  File "/usr/bin/calico-acl-manager", line 9, in <module>
    load_entry_point('calico==0.8', 'console_scripts', 'calico-acl-manager')()
  File "/usr/lib64/python2.7/site-packages/calico/acl_manager/acl_manager.py", line 38, in main
    plugin_address = config.get('global', 'PluginAddress')
  File "/usr/lib64/python2.7/ConfigParser.py", line 607, in get
ConfigParser.NoSectionError: No section: 'global'
1

# neutron-calico-agent ; echo $?
Traceback (most recent call last):
  File "/usr/bin/neutron-calico-agent", line 9, in <module>
    load_entry_point('calico==0.8', 'console_scripts', 'neutron-calico-agent')()
  File "/usr/lib64/python2.7/site-packages/calico/agent/calico_neutron_agent.py", line 102, in main
    logging_config.setup_logging(cfg.CONF)
TypeError: setup_logging() takes no arguments (1 given)
1
```

--- a/calico/felix/felix.py
+++ b/calico/felix/felix.py
@@ -25,6 +25,7 @@ import argparse
 import collections
 import logging
 import socket
+import sys
 import time
 import uuid
 import zmq
@@ -726,6 +727,7 @@ def main():
 
     except:
         log.exception("Felix exiting after uncaught exception")
+        sys.exit(1)
 
 if __name__ == "__main__":
     main()

